### PR TITLE
Remove iterator fields

### PR DESCRIPTION
**Pull request by gibson042**
Builds upon #56
Fixes #54

### CODE REVIEW

**Comment by littledan** [comment]
With this change, how should the information be accessed following a call to the `preceding` or `following` methods?

Would you be up for discussing this change in the Intl call? If so, please make a PR against the agenda at https://github.com/tc39/ecma402/blob/master/meetings/agenda-2018-11-26.md

**Comment by gibson042** [comment]
> With this change, how should the information be accessed following a call to the `preceding` or `following` methods?

From `next()` or a non-advancing analogue.

> Would you be up for discussing this change in the Intl call? If so, please make a PR against the agenda at https://github.com/tc39/ecma402/blob/master/meetings/agenda-2018-11-26.md

Absolutely. I'll add a point for the other issue as well.

File: README.md [hunk 1]
```diff
@@ -48,11 +48,20 @@ This method creates a new `%SegmentIterator%` over the input string, which will
...
 
 This class iterates over segment boundaries of a particular string.
+
+#### Iteration result data
+
+* `index` is the code point index immediately following the last found boundary.
+* `segment` is the substring between the last found boundary and its preceding boundary.
+* `breakType` is a value characterizing `segment` and its terminating boundary. Some particularly important values are:
+  * For `word` granularity, "word" for letter/number/ideograph segments vs. "none" for spaces/punctuation/etc.
+  * For `line` granularity, "hard" for mandatory line breaks such as follow U+000A LINE FEED (`\n`) vs. "soft" for line break opportunities such as follow U+0020 SPACE.
```
Comment (FrankYFTang) [inline README.md]: The `line` granularity is removed from the spec in Jan. So please remove this part.

**Comment by FrankYFTang** [comment]
I have similar questions:
1. What is the MOTIVATION to remove `get %SegmentIterator%.prototype.index` and `get %SegmentIterator%.prototype.breakType` ?
2. Since `preceding` and `following` return a boolean instead of an `IteratorResult` object, what will the `preceding` and `following`  usage looks like once we take this PR?

**Comment by littledan** [comment]
I've explained why it's important to keep these fields; see the April 2019 ECMA-402 notes for details. I'm not going to adopt this change.

**Pull request closed without merging**